### ISSUE TITLE
Add an argument to select the PHP binary/version to use

### ISSUE THREAD

**Issue by domsson** [matomo-org/matomo-log-analytics#215]
We're trying to run the `import_logs.py` on a server with several PHP versions installed, but where the `php` command (`/usr/bin/php`) is a symlink to `php4.4` and we do not have permissions to change that. We can easily run individual PHP scripts with other versions by running them with `php7.1 name-of-script.php` instead of using just `php`, but when using `import_logs.py`, it seems to run `php`, which gives the old version and hence leads to errors, aborting the script.

I looked through the command line arguments, hoping we could specify the PHP version to use, but failed to find one. Would it be possible to add such an argument? Or maybe there is a different way of achieving all this that I currently fail to see. Cheers!

**Comment by fdellwing** [comment]
Please test if changing the `phpBinary` in line 1041 to `php7.1` solves the issue.

**Comment by domsson** [comment]
Unaltered, the script gives this error message:

    Fatal error: `php /path-to-webroot/htdocs/matomo/misc/cron/updatetoken.php --piwik- 
       domain=sub.main.tld` failed with error: .
    Reponse code was: 255. You can alternatively run the importer using the --login and --password option

When changing `phpBinary` to `php7.1` as described above, this happens:

    Traceback (most recent call last):
      File "matomo/misc/log-analytics/import_logs.py", line 2612, in <module>
        config.init_token_auth()
      File "matomo/misc/log-analytics/import_logs.py", line 1087, in init_token_auth
        self.options.matomo_token_auth = self._get_token_auth()
      File "matomo/misc/log-analytics/import_logs.py", line 1072, in _get_token_auth
        credentials = open(filename, 'r').readline()
    IOError: [Errno 2] No such file or directory: 'X-Powered-By: PHP/7.1.17\r\nContent-type: text/html; 
    charset=UTF-8\r\n\r\n/path-to-webroot/htdocs/matomo/tmp/cache/token.php'

However, I think that's again due to the interpreter. Changing `phpBinary` to `php7.1-cli` yields this:

    0 lines parsed, 0 lines recorded, 0 records/sec (avg), 0 records/sec (current)
    Parsing log logs/access.log.16.gz...
    Fatal error: the selected log format doesn't include the hostname: you must specify the Matomo site 
    ID with the --idsite argument

Which, I assume, means that the script works fine, and I'm now facing an unrelated error.

In summary, for users like me (I can't change the default PHP version and the hoster is unwilling to do so) being able to hand in the `phpBinary` as command line argument would be very helpful. Until then, editing the script seems to do the trick.

**Comment by ryrun** [comment]
Can this please be fixed? Shouldn't be such a big problem :/ I need a phpbinary parameter, because "php" currently defaults to a much older version on my hosting server (plesk).

**Comment by sgiehl** [comment]
@ryrun Could you try passing login and password to the log importer as a workaround?  I'll have a look if we could update and merge the existing pull request to fix this issue
